docs/runners/sagemaker.md

```diff
@@ -0,0 +1,97 @@
+
+    ```bash
+    aws s3 mb s3://<bucket-name>
+    ```
+!!! note "IMPORTANT"
+
+    Regarding [the bucket and SageMaker permissions](https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning-ex-bucket.html):
+
+    - If you use the the term 'sagemaker' in the name of the bucket, SageMaker will automatically have the correct permissions to the access bucket.
+    - If you use any other name or existing bucket you will need to add a policy on the role that SageMaker uses to access the bucket. 
+
+
+You can then set this bucket as the `base_path` of your pipeline with the syntax: `s3://<bucket_name>/<path>`.
+
+### Running a pipeline with SageMaker
+
+
+Since compiling a sagemaker spec requires access to the AWS SageMaker API, you will need to be logged in to 
+AWS with a role that has all the required permissions to launch a SageMaker pipeline. 
+
+
+=== "Console"
+    
+    ```bash 
+    fondant run sagemaker <pipeline_ref> \
+     --role-arn $SAGEMAKER_ROLE_ARN 
+    ```
+    
+
+=== "Python"
+    
+    ```python
+    from fondant.pipeline.runner import SageMakerRunner
+    
+    runner = SageMakerRunner()
+    runner.run(
+        input=<path_to_pipeline>,
+        role_arn=<role_arn>,
+        pipeline_name=<pipeline_name>
+        instance_type=<instance_type>,
+    )
+    ```
+
+
+Once your pipeline is running you can monitor it using the SageMaker [Studio](https://aws.amazon.com/sagemaker/studio/).
+
+
+
+#### Using custom Fondant components on SageMaker
+
+SageMaker only supports images hosted on a private ECR registry. If you want to use custom Fondant components on SageMaker you will need to build and push them to your private ECR registry first. You can do this using the `fondant build` command.
+
+But first you need to login into Docker with valid ECR credentials more info [here](https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html):
+```bash
+aws ecr get-login-password --region <region> | docker login --username AWS --password-stdin <aws_account_id>.dkr.ecr.<region>.amazonaws.com
+```
+
+You will need to create a repository for you component first (one time operation):
+```bash
+ aws ecr create-repository --region <region> --repository-name <component_name>
+```
+
+Now you can use the `fondant build` [command](../components/publishing_components.md) (which uses Docker under the hood) to build and push your custom components to your private ECR registry:
+```bash
+fondant build <component dir> -t <aws_account_id>.dkr.ecr.<region>.amazonaws.com/<component_name>:<tag>
+```
+
+
+#### Assigning custom resources to the pipeline
+
+You can specify different kinds of `instance_type` while running your pipeline, you can see the available options [here](https://docs.aws.amazon.com/sagemaker/latest/dg/notebooks-available-instance-types.html). You may need to request a [quota increase](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for certain `instance_type`'s.
```

Comment:
Does this need to be done with the CLI, or do we support a `Resources` instance for Sagemaker as well?

Reply:
I added it: https://github.com/ml6team/fondant/pull/710